simulator/infra/ports/module.cpp

```diff
@@ -41,9 +47,64 @@ void Module::enable_logging_impl( const std::unordered_set<std::string>& names)
         c->enable_logging_impl( names);
 }
 
+void Module::module_dumping( pt::ptree* modules) const
+{
```

Comment:
This function does 3 things:
1. Dumps read ports
2. Dumps write ports
3. Dumps submodules

Could you please move each action do dedicated function?